pkg/flow/grpc-workflow-variables.go

```diff
@@ -443,6 +443,71 @@ func (flow *flow) SetVariable(ctx context.Context, vrefc *ent.VarRefClient, vdat
 	return vdata, newVar, err
 }
 
+func (flow *flow) DeleteVarialbe(ctx context.Context, vrefc *ent.VarRefClient, vdatac *ent.VarDataClient, q varQuerier, key string, data []byte, vMimeType string, thread bool) (*ent.VarData, bool, error) {
```

Comment:
I don't think this is quite right. Variables were split into `varref` and `vardata` so that multiple references to a shared variable could be possible. That makes the logic a bit more complex. If you look at `DeleteWorkflowVariable` for example, you'll see in there we delete the ref, count the number of refs pointing at the data and only delete the data if the count has dropped to zero. I can't see any similar logic happening here.